### PR TITLE
login and registration added

### PR DESCRIPTION
**Pull request by Rakshita-7**
Login and Registration added

### CODE REVIEW

File: client/src/components/OTPComponent.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const OTPComponent = ({ email, onOtpVerified }) => {
+  const [otp, setOtp] = useState('');
+  const [isOtpSent, setIsOtpSent] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  // Send OTP to the user's email
+  const handleSendOtp = async () => {
+    setLoading(true);
+    try {
+      const response = await axios.post('http://127.0.0.1:8000/send-otp', { email });
```
Comment (sh1v4mr4j) [inline client/src/components/OTPComponent.js:14]: minor change: we are using fetch instead of axios

Comment (sh1v4mr4j) [reply client/src/components/OTPComponent.js:14]: This call can be put to services

File: client/src/pages/LoginPage.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  };
+
+  // Handle form submission
+  const onFinish = (values) => {
+    const { username, password, email } = values;
+    console.log('Login Form Values:', values);
+
+    // Define the login URL based on the selected role
+    const loginUrl = role === 'patient'
+      ? 'http://127.0.0.1:8000/patients/patientLogin'
```
Comment (sh1v4mr4j) [inline client/src/pages/LoginPage.jsx]: can put this in endpoint.js

File: client/src/pages/LoginPage.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        <Form.Item
+          name="password"
+          label="Password"
+          rules={[{ required: true, message: 'Please input your password!' }]}
+        >
+          <Input.Password prefix={<LockOutlined />} placeholder="Password" />
+        </Form.Item>
+
+        <Form.Item>
+          <Checkbox>Remember me</Checkbox>
```
Comment (sh1v4mr4j) [inline client/src/pages/LoginPage.jsx]: Does this work?

File: client/src/pages/LoginPage.jsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            message: 'Login Successful',
+            description: `Welcome, ${role.charAt(0).toUpperCase() + role.slice(1)}!`,
+            duration: 3,
+          });
+          navigate(`/${role.charAt(0).toUpperCase()+role.slice(1)}Page`);
+        }
+      })
+      .catch((error) => {
+        // If there's any error during the fetch call
+        console.error('Error:', error);
```
Comment (sh1v4mr4j) [inline client/src/pages/LoginPage.jsx]: throw error instead

**Comment by sh1v4mr4j** [comment]
can you let me know where are we setting the user context?

File: client/src/components/OTPComponent.js (hunk 1 above)
Comment (Rakshita-7) [reply client/src/components/OTPComponent.js:14]: Yes, wip. shall update

File: client/src/pages/LoginPage.jsx (hunk 2 above)
Comment (Rakshita-7) [reply client/src/pages/LoginPage.jsx]: noted. Shall push in next PR

File: client/src/pages/LoginPage.jsx (hunk 3 above)
Comment (Rakshita-7) [reply client/src/pages/LoginPage.jsx]: WIP

**Pull request closed without merging**